src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVReferenceUtils.java

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.hellbender.tools.spark.sv.utils;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.spark.HashPartitioner;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.utils.HopscotchMap;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class SVReferenceUtils {
+
+    public static final int REF_RECORD_LEN = 10000;
+    // assuming we have ~1Gb/core, we can process ~1M kmers per partition
+    public static final int REF_RECORDS_PER_PARTITION = 1024*1024 / REF_RECORD_LEN;
+
+    public static List<String> getRefNames(final SAMFileHeader header) {
+        return header.getSequenceDictionary().getSequences().stream()
+                .map(SAMSequenceRecord::getSequenceName).collect(Collectors.toList());
+    }
+
+    /**
+     * Create an RDD from the reference sequences.
+     * The reference sequences are transformed into a single, large collection of byte arrays. The collection is then
+     * parallelized into an RDD.
+     * Each contig that exceeds a size given by REF_RECORD_LEN is broken into a series of REF_RECORD_LEN chunks with a
+     * K-1 base overlap between successive chunks. (I.e., for K=63, the last 62 bases in chunk n match the first 62
+     * bases in chunk n+1) so that we don't miss any kmers due to the chunking -- we can just kmerize each record
+     * independently.
+     */
+    public static JavaRDD<byte[]> getRefRDD(final JavaSparkContext ctx,
```

Comment:
Maybe this could be moved to `SparkUtils`... may be useful outside the SV code...

Reply:
This method is not really tested well I am hesitated to move it there. Beside I am not the author.
What usage do you have in mind?

Reply:
I have an usage case where I need a tool iterating over kmers in the reference: although still in development, I have a reference walker by kmers, and at some point it would be nice to have also a spark-based reference walker.

Anyway, it was just a suggestion. For me, it is just simpler to look for useful methods in the utils classes related to the stuff I'm looking for (e.g., `CigarUtils` for cigar, `SparkUtils`for RDD related stuff, etc.

Reply:
@magicDGS in that case (considering your tool is still under dev), I intend to keep it here and get more test coverage when you think you are ready. Is that OK?

Reply:
It's perfectly fine, I'm just worried that I forget that this is here and implement my own method. Usually I look at the `SparkUtils` code for this kind of things, and I do not use to look at the SV code at all except in some PRs with functionality that I think that I can use as a third-party developer...

Reply:
@magicDGS . In that case, I'll try to add tests and move it to SparkUtils. In the meantime, if you find other methods in `SVUtils` and `SvCigarUtils` that are useful to you, let me know.